distributed/utils.py

```diff
@@ -371,29 +371,39 @@ def in_async_call(loop, default=False):
         return False
 
 
-def sync(loop, func, *args, callback_timeout=None, **kwargs):
+def sync(
+    loop: IOLoop,
+    func: Callable[..., Awaitable[T] | asyncio.Future[T]],
+    *args: AnyType,
+    callback_timeout: str | float | timedelta | None = None,
+    **kwargs: AnyType,
+) -> T:
     """
     Run coroutine in loop running in separate thread.
     """
-    callback_timeout = _parse_timedelta(callback_timeout, "s")
-    if loop.asyncio_loop.is_closed():
+    timeout = _parse_timedelta(callback_timeout, "s")
+    if loop.asyncio_loop.is_closed():  # type: ignore[attr-defined]
         raise RuntimeError("IOLoop is closed")
 
     e = threading.Event()
     main_tid = threading.get_ident()
-    result = error = future = None  # set up non-locals
+
+    # set up non-locals
+    result: T
+    error = None
+    future: asyncio.Future[T] | asyncio.Task[T]
```

Comment:
Task is an instance of Future so 
```suggestion
    future: asyncio.Future[T]
```

Reply:
Is it? The docs state
> [asyncio.Task](https://docs.python.org/3/library/asyncio-task.html#asyncio.Task) inherits from [Future](https://docs.python.org/3/library/asyncio-future.html#asyncio.Future) all of its APIs except [Future.set_result()](https://docs.python.org/3/library/asyncio-future.html#asyncio.Future.set_result) and [Future.set_exception()](https://docs.python.org/3/library/asyncio-future.html#asyncio.Future.set_exception).

...but I guess it's close enough for our purposes.

Reply:
For papertrail: https://github.com/python/typeshed/blob/main/stdlib/asyncio/tasks.pyi#L382

Typing considers them the same.